components/director/internal/domain/formation/repository.go

```diff
@@ -147,8 +147,21 @@ func (r *repository) Update(ctx context.Context, model *model.Formation) error {
 	if model == nil {
 		return apperrors.NewInternalError("model can not be empty")
 	}
-	log.C(ctx).Debugf("Updating formation with ID: %q and name: %q...", model.ID, model.Name)
-	return r.updater.UpdateSingleGlobal(ctx, r.conv.ToEntity(model))
+	newEntity := r.conv.ToEntity(model)
+
+	var retrievedEntity Entity
+	if err := r.globalGetter.GetGlobal(ctx, repo.Conditions{repo.NewEqualCondition("id", model.ID)}, repo.NoOrderBy, &retrievedEntity); err != nil {
+		return err
+	}
+
+	if retrievedEntity.State != newEntity.State {
```

Comment:
What if we are in an error state and after executing a notification the response contains error that is different than the previous one? Shouldn't we update the timestamp in this case also? The `last_notification_sent_timestamp` will be updated with the execution of the notification, but the  `last_state_change_timestamp` will not despite we received a response.

Reply:
As far as I understand we only want to update this timestamp(`last_state_change_timestamp`) in case the state is changed? So in the scenario that you described the state will be the same.

Reply:
discussed offline and the code is adjusted.